### PR TITLE
Add recurring command

### LINKED ISSUE #91: add recurring task feature

### PR DESCRIPTION
**Pull request by umergta**
- basic functionality out.
issues:
- letting the user add a task with deadline before now (as i will add the day/week/time accordingly upon loading to load the next recurring task)

- Will add test cases for recur commands later
- Will add some javadocs for public methods later

### CODE REVIEW

File: src/main/java/seedu/module/logic/commands/RecurCommand.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import seedu.module.model.task.Deadline;
+import seedu.module.model.task.DoneStatus;
+import seedu.module.model.task.Recurrence;
+import seedu.module.model.task.Task;
+
+
+/**
+ * Makes a task repeat at a given interval of
+ */
+public class RecurCommand extends Command {
```
Comment (yhtMinceraft1010X) [inline src/main/java/seedu/module/logic/commands/RecurCommand.java:21]: I have a few questions:
1) Wouldn't the edit command suffice for changing the recurrence of a task?
2) If we decide to keep the recur command, is there a way to remove recurrence without using edit command? I think users may be confused as to why the way to remove recurrence is not obvious.

**Review by yhtMinceraft1010X — COMMENTED** [review]
Everything else looks good as of now.

File: src/main/java/seedu/module/logic/commands/RecurCommand.java (hunk 1 above)
Comment (yhtMinceraft1010X) [reply src/main/java/seedu/module/logic/commands/RecurCommand.java:21]: Actually, for 1, scratch the question. We also have a tag command and edit command that can handle tag changes.

**Review by AiwassPrime — COMMENTED** [review]
Some more test for recur and recurParser

**Comment by umergta** [comment]
Merged upstream repo into current repo for recurring-command:

Failing testcases (that have been commented out):
1. JsonSerializableModuleBookTest: toModelType_typicalTasksFile_success()

2. StorageManagerTest: moduleBookReadSave()

3. JsonModuleBookStorageTest: readAndSaveModuleBook_allInOrder_success()

File: src/main/java/seedu/module/logic/commands/RecurCommand.java (hunk 1 above)
Comment (umergta) [reply src/main/java/seedu/module/logic/commands/RecurCommand.java:21]: Yes. I think there should be a notRecur command to remove the recurrence of the Task. I think i will work on that Monday after clearing other assignments haha

**Review by yhtMinceraft1010X — COMMENTED** [review]
LGTM

**Pull request merged**